### PR TITLE
Add faq app

### LINKED ISSUE #25: Add SmartFAQ
add a new basic query that should be able to search by term input and language to fetch static FAQ

### PR DESCRIPTION
**Pull request by jalbertsr**
Close #25

### CODE REVIEW

File: src/smartFAQ/queries/SmartFAQ.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  query: string,
+  language: string,
+};
+
+export default {
+  type: GraphQLSmartFAQ,
+  description: 'Smart FAQ help center',
+  args: {
+    query: {
+      type: GraphQLString,
```
Comment (mrtnzlml) [inline src/smartFAQ/queries/SmartFAQ.js]: So is query required or not? It looks like it should be required (according to the tests).

File: src/smartFAQ/types/outputs/Results.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+// @flow
```
Comment (mrtnzlml) [inline src/smartFAQ/types/outputs/Results.js]: https://github.com/kiwicom/graphql/#design-style-guides

File: src/common/services/JsonFetcher.js [hunk 3]
```diff
@@ -6,8 +6,8 @@ import { ProxiedError } from './errors/ProxiedError';
 
 export async function fetchJson(
   url: string,
-  method: string = 'GET',
-  options?: Object = {},
+  method: string,
```
Comment (michalsanger) [inline src/common/services/JsonFetcher.js]: Why did you remove the default values?

File: src/common/services/JsonFetcher.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ export async function fetchJson(
 
   Logger.info(`${method} ${url}`);
 
-  const response = await fetch(url, reqOptions);
+  const response = await fetch((url: string), (reqOptions: Object));
```
Comment (michalsanger) [inline src/common/services/JsonFetcher.js]: Are those types here adding something valuable? The types of `url` and `reqOptions` are known I'd say.

File: src/RootQuery.js [hunk 5]
```diff
@@ -29,5 +30,6 @@ export default new GraphQLObjectType({
     currentUser: CurrentUser,
     hotel: Hotel,
     hotelCities: HotelCities,
+    searchSmartFAQ: SearchSmartFAQ,
```
Comment (michalsanger) [inline src/RootQuery.js]: Not sure with the naming. "Smart FAQ" is name of the project, but the query name should correspond with the entities and I'd say. Also so far we hame naming style, that query returning list is prefixed with "all" (`allBookings`, `allFlights`). 

So my suggestion is `allFAQs` and `query` renamed to `search` and being optional, because I think there will be use case just for listing FAQs without any search. And speaking about listing, pagination should be supported and we use Relay standard for that so `FaqConnection` should be returned.

File: src/smartFAQ/queries/__test__/SmartFAQ.test.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import SmartFAQDatasetEN from '../../datasets/smartFAQ-EN.json';
+import SmartFAQDatasetES from '../../datasets/smartFAQ-ES.json';
+import SmartFAQDefault from '../../datasets/smartFAQ-default.json';
+
+beforeEach(() => {
+  RestApiMock.onGet(
+    'https://api.skypicker.com/knowledgebase/api/v1/search?q=price',
+  ).replyWithData(SmartFAQDatasetEN);
+  RestApiMock.onGet(
+    'https://api.skypicker.com/knowledgebase/api/v1/search?q=maleta',
```
Comment (michalsanger) [inline src/smartFAQ/queries/__test__/SmartFAQ.test.js]: The language is not in URL?

File: src/smartFAQ/queries/__test__/SmartFAQ.test.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+          }
+        }
+      }
+    }`;
+    expect(await graphql(resultsQuery)).toMatchSnapshot();
+  });
+
+  it('should work without language and empty query', async () => {
+    const resultsQuery = `{
+      searchSmartFAQ(query: ""){
```
Comment (michalsanger) [inline src/smartFAQ/queries/__test__/SmartFAQ.test.js]: I don't like empty string as a param, better make the param optional

File: src/smartFAQ/dataloaders/__test__/ApiSanitizer.test.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+// @flow
+
+import { sanitizeSmartFAQ } from '../ApiSanitizer';
+import smartFAQResponse from '../../datasets/smartFAQ-EN.json';
+
+describe('Sanitize list of results', () => {
+  it('should work', async () => {
+    const smartFAQResults = smartFAQResponse.map(sanitizeSmartFAQ);
```
Comment (michalsanger) [inline src/smartFAQ/dataloaders/__test__/ApiSanitizer.test.js]: Keep the snapshot file small and readable, test only one item or more if there is any edge case that makes sense.

File: src/smartFAQ/queries/__test__/SmartFAQ.test.js (hunk 6 above)
Comment (jalbertsr) [reply src/smartFAQ/queries/__test__/SmartFAQ.test.js]: The language is set in the HTTP headers, as it is explained in the api docs: https://gitlab.skypicker.com/django-team/knowledgebase#api

File: src/common/services/JsonFetcher.js (hunk 3 above)
Comment (jalbertsr) [reply src/common/services/JsonFetcher.js]: I changed  `fetchJson(url.format(urlObject))` to `fetchJson(url.format(urlObject), 'GET', { headers })`  so now this default parameters are never reached, then the branch coverage decreases below 80%.

File: src/smartFAQ/queries/AllFAQs.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+import type { GraphqlContextType } from '../../common/services/GraphqlContext';
+
+const { connectionType: FaqConnection } = connectionDefinitions({
+  nodeType: GraphQLSmartFAQ,
+});
+
+export default {
+  type: FaqConnection,
+  description: 'Smart FAQ help center',
```
Comment (michalsanger) [inline src/smartFAQ/queries/AllFAQs.js]: "Search for Frequently Asked Questions"

File: src/smartFAQ/types/outputs/Results.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+      resolve: ({ title }: ItemsType): string => title,
+    },
+    icon: {
+      type: GraphQLString,
+      description: 'Icon type of the item',
+      resolve: ({ icon }: ItemsType): string => icon,
+    },
+    style: {
+      type: GraphQLString,
+      description: 'Style of the item',
```
Comment (michalsanger) [inline src/smartFAQ/types/outputs/Results.js]: Not sure what does this mean, can you explain the `style` more in the description please?
